src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,300 @@
+        End Function
+
+        Private Function AddAnalysisPoint(syntaxForSpan As VisualBasicSyntaxNode, statementFactory As SyntheticBoundNodeFactory) As BoundStatement
+            ' Add an entry in the spans array.
+
+            Dim location As Location = syntaxForSpan.GetLocation()
+            Dim spanPosition As FileLinePositionSpan = location.GetMappedLineSpan()
+            Dim path As String = spanPosition.Path
+            If path.Length = 0 Then
+                path = syntaxForSpan.SyntaxTree.FilePath
+            End If
+
+            Dim spansIndex As Integer = _spansBuilder.Count
+            _spansBuilder.Add(New SourceSpan(_debugDocumentProvider.Invoke(path, ""), spanPosition.StartLinePosition.Line, spanPosition.StartLinePosition.Character, spanPosition.EndLinePosition.Line, spanPosition.EndLinePosition.Character))
+
+            ' Generate "_payload(pointIndex) = True".
+
+            Dim payloadCell As BoundArrayAccess = statementFactory.ArrayAccess(statementFactory.Local(_methodPayload, False), True, statementFactory.Literal(spansIndex))
+            Return statementFactory.Assignment(payloadCell, statementFactory.Literal(True))
+        End Function
+
+        Private Shared Function SyntaxForSpan(statement As BoundStatement) As VisualBasicSyntaxNode
+            Select Case statement.Kind
+                Case BoundKind.IfStatement
+                    Return DirectCast(statement, BoundIfStatement).Condition.Syntax
+                Case BoundKind.WhileStatement
+                    Return DirectCast(statement, BoundWhileStatement).Condition.Syntax
+                Case BoundKind.ForEachStatement
+                    Return DirectCast(statement, BoundForEachStatement).Collection.Syntax
+                Case BoundKind.DoLoopStatement
+                    Return DirectCast(statement, BoundDoLoopStatement).ConditionOpt.Syntax
+                Case BoundKind.UsingStatement
+                    Dim usingStatement As BoundUsingStatement = DirectCast(statement, BoundUsingStatement)
+                    Return If(usingStatement.ResourceExpressionOpt IsNot Nothing, DirectCast(usingStatement.ResourceExpressionOpt, BoundNode), usingStatement).Syntax
+                Case BoundKind.SyncLockStatement
+                    Return DirectCast(statement, BoundSyncLockStatement).LockExpression.Syntax
+                Case BoundKind.SelectStatement
+                    Return DirectCast(statement, BoundSelectStatement).ExpressionStatement.Expression.Syntax
+            End Select
+
+            Return statement.Syntax
+        End Function
+
+        Private Shared Function MethodHasExplicitBlock(method As MethodSymbol) As Boolean
+            Dim asSourceMethod As SourceMethodSymbol = TryCast(method.OriginalDefinition, SourceMethodSymbol)
+            If asSourceMethod IsNot Nothing Then
+                Return TypeOf asSourceMethod.Syntax Is MethodBlockBaseSyntax
```

Comment:
Missed this during previous pass. TypeOf is always false for ```Nothing```, so the null check is redundant here.